src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -182,11 +197,39 @@ public async Task<bool> TryRefreshAsync()
             {
                 await RefreshAsync().ConfigureAwait(false);
             }
-            catch (Exception e) when (
-                e is KeyVaultReferenceException ||
-                e is RequestFailedException ||
-                ((e as AggregateException)?.InnerExceptions?.All(e => e is RequestFailedException) ?? false) ||
-                e is OperationCanceledException)
+            catch (RequestFailedException e)
+            {
+                if (e.Status == (int)HttpStatusCode.Unauthorized || e.Status == (int)HttpStatusCode.Forbidden)
+                {
+                    _logger?.LogWarning(e, "Refresh operation failed due to an authentication error.");
+                }
+                else
+                {
+                    _logger?.LogWarning(e, "Refresh operation failed due to an error.");
+                }
+
+                return false;
+            }
+            catch (AggregateException e) when (e?.InnerExceptions?.All(e => e is RequestFailedException) ?? false)
+            {
+                if (e.InnerExceptions.Any(exception => (exception is RequestFailedException ex) 
+                                                        && (ex.Status == (int)HttpStatusCode.Unauthorized || ex.Status == (int)HttpStatusCode.Forbidden)))
+                {
+                    _logger?.LogWarning(e, "Refresh operation failed due to an authentication error.");
+                }
+                else
+                {
+                    _logger?.LogWarning(e, "Refresh operation failed due to an error.");
+                }
+
+                return false;
+            }
+            catch (KeyVaultReferenceException e)
+            {
+                _logger?.LogWarning(e, "Refresh operation failed while resolving a Key Vault reference.");
+                return false;
+            }
+            catch (OperationCanceledException)
             {
                 return false;

```

Comment:
Missing a log?

Reply:
Since we don't support cancellation token in `TryRefreshAsync` flow, we shouldn't be handling `OperationCanceledException` in the first place. I was on the fence about adding a log here or adding the log later when we work on cancellation token support. But I guess it doesn't hurt to add it now.